cheroot/test/test_conn.py

```diff
@@ -121,7 +125,39 @@ def _timeout(req, resp):
     wsgi_server.timeout = timeout
     wsgi_server.server_client = wsgi_server_client
     wsgi_server.keep_alive_conn_limit = 2
-    return wsgi_server
+
+    # patch the error_log calls of the server instance
+    class ErrorLog:  # noqa: D200
+        """
+        Mock class to access the server error_log calls made by the server.
+        """
+
+        FuncCall = namedtuple('ErrorLogCall', ['msg', 'level', 'traceback'])
+
+        def __init__(self):
+            self.calls = []
+            # to be used the the teardown validation
+            self.ignored_msgs = []
+
+        def __call__(self, msg='', level=logging.INFO, traceback=False):
+            if traceback:
+                tblines = traceback_.format_exc()
+            else:
+                tblines = ''
+            self.calls.append(ErrorLog.FuncCall(msg, level, tblines))
```

Comment:
Looks like this should go to the module level, no need to keep it inside the fixture function + there would be no need for noqa too.

Reply:
I personally prefer to keep these close to where they are relevant for test. Otherwise they are just _noise_ for the rest of the tests (ideally having _only_ tests at the top level of the module), but I must admit that this _philosophy_ comes from other projects (I believe I got this from pyramid) where there is a clear distinction in regular code standards and test code in terms of nesting functionality and imports inside tests for example .  This indeed becomes a problem conforming for pydoc style but having the same standard in the doctrings for tests that are never going to be parsed by any tool other than the eyes of the developer doesn't make a lot of sense to me, other than been dogmatic to conform to the linter. 

Having said that... if you _really_ want that I can make it  happen.

Reply:
This request doesn't only come from the linter violations. The true origin is that such things make noise inside the test itself making it harder to consume its contents. I usually prefer to have such things in fixtures so that the test only operates things on one level of abstraction.

Reply:
Or in this case, it'd be another fixture dependency maybe...